packages/oracle-manager/test/integration/modules/NodeModule.test.ts

```diff
@@ -0,0 +1,97 @@
+import assert from 'assert/strict';
+import assertBn from '@synthetixio/core-utils/utils/assertions/assert-bignumber';
+import { ethers } from 'ethers';
+
+import { bootstrapWithNodes } from '../bootstrap';
+import NodeTypes from '../mixins/Node.types';
+import NodeOperations from '../mixins/Node.operations';
+
+describe('NodeModule', function () {
+  const { signers, systems, nodeId1, nodeId2, nodeId3, abi } = bootstrapWithNodes();
+
+  let owner: ethers.Signer;
+  let parents: string[];
+
+  before('identify signers', async () => {
+    [owner] = signers();
+    parents = [nodeId1(), nodeId2(), nodeId3()];
+  });
+
+  it('make sure mock aggregator node is set up', async () => {
+    const node = await systems().Core.connect(owner).getNode(nodeId1());
+    assert.notEqual(node.nodeType, NodeTypes.NONE);
+  });
+
+  it('Test price on leaf nodes', async () => {
+    let priceData = await systems().Core.connect(owner).process(nodeId1());
+    assertBn.equal(priceData.price, ethers.utils.parseEther('1'));
+
+    priceData = await systems().Core.connect(owner).process(nodeId2());
+    assertBn.equal(priceData.price, ethers.utils.parseEther('0.9'));
+  });
+
+  it('register a max reducer', async () => {
```

Comment:
specific tests regarding the reducers should go in their own file which specifically tests the library functions outside of the interference of the Node factory. I can show you how to do this later if you want, its actually not too bad.